src/Mailgun/Api/Message.php

```diff
@@ -129,4 +144,33 @@ private function prepareFile($fieldName, array $filePath)
             'filename' => $filename,
         ];
     }
+
+    /**
+     * Prepare multipart parameters. Make sure each POST parameter is splitted into an array with 'name' and 'content' keys.
+     *
+     * @param array $params
+     *
+     * @return array
+     */
+    private function prepareMultipartParameters(array $params)
+    {
+        $postDataMultipart = [];
+        foreach ($params as $key => $value) {
+            if (is_array($value)) {
+                foreach ($value as $subValue) {
```

Comment:
hm.. We can improve this. But recursion is wrong because we do not allow a tree with depth more that 2.